interview-questions.md

```diff
@@ -8,44 +8,72 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What does calling super do in a Ruby class?
 
-Your answer:
-
-Researched answer:
+Your answer: Super is used when creating a child class and it is used to reference an attribute that is being inherited from the parent class.
+
+For example:
+```rb
+# create parent class
+class Car
+    def initialize (color, transmission, engine_size)
+        @color = color
+        @transmission = transmission
+        @engine_size = engine_size
+    end
+end
+
+# create child class 
+class Supercar < Car
+    def initialize (color, transmission, engine_size, max_speed)
+        super(color)
+        super(transmission)
+        super(engine_size)
+        @max_speed = max_speed
+    end
+end 
+```
+In this example the child class inherits the attributes and refers to the methods of the parent class in order to use them in the same way, but can expand beyond the parent class by adding @max_speed attribute
+
+Researched answer: You can also use methods from the parent class by just using super or super(). The first way, super, assumes you are using the same arguments as what the parent class uses, so if you use a different number of arguments it will throw a "wrong number of arguments" error. When you use super() it allows you to pass a different number of arguments than that of the parent class. 
 
 2. What is a gem?
 
-Your answer:
+Your answer: A gem is an addon to the Ruby language
 
-Researched answer:
+Researched answer: Gems are software packages that contain ruby applications to extend or modify ruby functionality. RSpec is a gem as well as Rails. 
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is a database in which the tables of the database can have relationships (connections) to one another. So in a video store db (yeah I said Blockbuster) I can have a table of customers and a table of movies. Then I can connect a specific movie as rented out to a specific customer. I know that there are dbs that are not relational but I don't remember their name. Essentially they are data buckets I believe MongoDB is one such non-relational type of db. 
 
-Researched answer:
+Researched answer: Haha so I guess if they aren't relational they are called non-relational db. Logic for once. NoSQL dbs (as they are also known) do not use tables, rows, columns, keys. Instead they storage of these databases is optimized for the specific usage of the db and the data being stored there. 
```

Comment:
There are relational and non-relational databases, but there are also many other types of databases out there!

Reply:
I guess it is time for even more research LOL